compiler/tools/fpp-to-cpp/src/main/scala/fpp-to-cpp.scala

```diff
@@ -40,11 +40,16 @@ object FPPToCpp {
         Parser.parseFile (Parser.transUnit) (None) _
       )
       a <- CheckSemantics.tuList(a, tulFiles ++ tulImports)
-      s <- ComputeCppFiles.visitList(
-        CppWriterState(a),
-        tulFiles,
-        ComputeCppFiles.transUnit
-      )
+      s <- if options.template then ComputeImplCppFiles.visitList(
```

Comment:
Let’s match on a mode variable here. We can add code to CppWriter that converts the boolean flags to the mode variable. Then the tools that need this code can use it.

Reply:
This will also let us extend the code when we add unit test code gen.